projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx

```diff
@@ -142,7 +146,7 @@ class SearchApp extends Component {
 	};
 
 	hasActiveQuery() {
-		return this.props.searchQuery !== '' || this.props.hasFilters;
+		return this.props.searchQuery !== '' || this.props.hasFilters || hasSearchQueryStringKey();
```

Comment:
Could we push `hasSearchQueryStringKey` into the Redux state?

I'd like for us to refrain from directly accessing `window.location` so we can rely purely on Redux as the source of truth. (I'm okay having `getResultFormat` in the component for the time being since it's just a temporary measure.)

Reply:
Working on it 👍 

I tried having a separate boolean in state and it got complicated - had to keep it in sync with the searchQuery and the modal open/closed state. After a bit of tinkering, I found the best answer was to throw away all my code and keep it simple:

`searchQuery` === null - search closed
`searchQuery` === '' - search open without a query

...so I've made some changes to handle the `null` value for the search query.